app/helpers/react_components/mentoring/representations/with_feedback.rb

```diff
@@ -0,0 +1,83 @@
+module ReactComponents
+  module Mentoring
+    module Representations
+      class WithFeedback < ReactComponent
+        initialize_with :mentor, :params
+
+        def to_s
+          super(
+            "mentoring-representations-with-feedback",
+            {
+              representations_request:,
+              tracks_request:,
+              links:,
+              sort_options: SORT_OPTIONS,
+              representations_without_feedback_count:,
+              is_introducer_hidden:
+            }
+          )
+        end
+
+        private
+        def representations_request
+          {
+            endpoint: Exercism::Routes.with_feedback_api_mentoring_representations_url,
+            query: representations_request_params,
+            options: {
+              initial_data: representations,
+              stale_time: 5000 # milliseconds
+            }
+          }
+        end
+
+        memoize
+        def representations_request_params
+          {
+            criteria: params.fetch(:criteria, ''),
```

Comment:
I still don't think we need the empty string thing here btw.

Reply:
We currently do, but we should be fixing it in typescript instead.